### PR TITLE
fix(is_enterprise): check command output applying logic

### PR DESCRIPTION
**Pull request by fruch**
since in cases we might have failure like the following:

```
Could not retrieve mirrorlist http://mirrorlist.centos.org/?release=7&arch=x86_64
    &repo=os&infra=stock error was

14: curl#7 - "Failed connect to mirrorlist.centos.org:80; Operation now in progress"
```

we wrong assumed this should be enterpise and tried to install enterpise artifacts (which didn't existed)

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -716,16 +716,21 @@ def start_network_interface(self, interface_name="eth1"):
...
+    def is_enterprise(self) -> bool | None:
+        _is_enterprise = None
         if self.distro.is_rhel_like:
-            result = self.remoter.sudo("yum search scylla-enterprise 2>&1", ignore_status=True).stdout
-            if 'One of the configured repositories failed (Extra Packages for Enterprise Linux 7 - x86_64)' in result:
-                return "enterprise" in self.remoter.sudo("cat /etc/yum.repos.d/scylla.repo").stdout
-            return "scylla-enterprise.x86_64" in result or "No matches found" not in result
+            result = self.remoter.sudo("yum search scylla-enterprise 2>&1", ignore_status=True)
+            if result.ok:
+                _is_enterprise = "scylla-enterprise.x86_64" in result.stdout or "No matches found" not in result.stdout
```
Comment (vponomaryov) [inline sdcm/cluster.py:724]: What about `enterprise` on `Arm`?

Comment (fruch) [reply sdcm/cluster.py:724]: I guess the other logic would cover it (i.e. No matches)

I just moved it around, the already available logic.

**Review by juliayakovlev — APPROVED** [review]

**Review by soyacz — APPROVED** [review]
LGTM

**Review by vponomaryov — APPROVED** [review]
LGTM

**Pull request merged**